pkg/cluster/executor/ssh.go

```diff
@@ -81,28 +98,41 @@ type (
 	}
 )
 
-var _ Executor = &SSHExecutor{}
+var _ Executor = &EasySSHExecutor{}
+var _ Executor = &NativeSSHExecutor{}
 
 // NewSSHExecutor create a ssh executor.
-func NewSSHExecutor(c SSHConfig, sudo bool) *SSHExecutor {
-	e := new(SSHExecutor)
-	e.Initialize(c)
-	e.Locale = "C" // default locale, hard coded for now
-	e.Sudo = sudo
-	return e
-}
-
-// Initialize builds and initializes a SSHExecutor
-func (e *SSHExecutor) Initialize(config SSHConfig) {
+func NewSSHExecutor(c SSHConfig, sudo bool, native bool) Executor {
 	// set default values
-	if config.Port <= 0 {
-		config.Port = 22
+	if c.Port <= 0 {
+		c.Port = 22
 	}
 
-	if config.Timeout == 0 {
-		config.Timeout = time.Second * 5 // default timeout is 5 sec
+	if c.Timeout == 0 {
+		c.Timeout = time.Second * 5 // default timeout is 5 sec
 	}
 
+	if native {
+		e := &NativeSSHExecutor{
+			Config: &c,
+			Locale: "C",
+			Sudo:   sudo,
+		}
+		if c.Password != "" || (c.KeyFile != "" && c.Passphrase != "") {
+			_, _, e.ConnectionTestResult = e.Execute(connectionTestCommand, false, executeDefaultTimeout)
```

Comment:
I think we shouldn't try to connect remote when we create the instance of `SSHExecutor`.

Reply:
If we don't try it this time when the user actually executes the command, it may block as long as the command timeout value since the wrong prompt of ssh. What's worse is that there will be no possibility to get clear what happened at that time.

For example, the user can change the ssh prompt to:
```
root@172.16.5.141's secret:
```
But the `sshpass` may expect it be:
```
root@172.16.5.141's password:
```
And then the command will hang forever (until receiving the kill signal).